engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -1,9 +1,32 @@
+require 'fileutils'
+
 module InstanceVerification
-  def self.update_cache(remote_ip, system_login, product_id, is_byos)
+  def self.update_cache(remote_ip, system_login, product_id, _is_byos)
+    # TODO: BYOS scenario
+    # to be addressed on a different PR
+    cache_config_data = InstanceVerification.cache_config
     cache_key = [remote_ip, system_login, product_id].join('-')
     # caches verification result to be used by zypper auth plugin
-    expire_cache_time = is_byos ? 24.hours : 20.minutes
-    Rails.cache.write(cache_key, true, expires_in: expire_cache_time)
+    InstanceVerification.write_cache_file(
+      cache_config_data['REPOSITORY_CLIENT_CACHE_DIRECTORY'],
+      cache_key
+    )
```

Comment:
But the implementation details state that the registry path should also be updated, this only updates the repository path.

Reply:
yes, that is done in a different PR

Reply:
This PR changes the cache of the repositories as it was to the new cache system

Reply:
The way you are building on top of each other is not really working, see my comment in #1141 . And if this is intended to change the current cache behavior then  setting the key related to registry in `config/application.rb` shouldn't be there.

I appreciate your effort in separating things to keep them smaller. But the unit of what is getting submitted should be consistent.

Reply:
Fixed, only repositories cache on the cache config file and cache test file on this PR and
added the registry lines on #1141